_targets.R

```diff
@@ -0,0 +1,46 @@
+library(targets)
+source("1_fetch/src/fetch_data.R")
+source("2_process/src/process_data.R")
+source("3_visualize/src/visualize_data.R")
+
+list(
+  # Get the data from ScienceBase
+  tar_target(
+    model_RMSEs_csv,
+    fetch_sb_file(sb_id = "5d925066e4b0c4f70d0d0599",
+                  name = "me_RMSE.csv",
+                  output_file = "1_fetch/out/model_RMSEs.csv"),
+    packages = c("sbtools"),
+    format = "file",
+  ), 
+  # Prepare the data for plotting
+  tar_target(
+    eval_data,
+    process_data(mendota_file = model_RMSEs_csv),
+    packages = c("dplyr", "stringr"),
+  ),
+  # Prepare the data for plotting
+  tar_target(
+    model_summary_results_csv,
+    save_summary_results(eval_data = eval_data,
+                 output_file = "2_process/out/model_summary_results.csv"),
+    packages = c("dplyr", "stringr"),
+    format = "file",
+  ),
```

Comment:
No change needed, as this approach (of calling a function that writes the csv and returns the filepath) works great, but just wanted to alert  you to another option for targets that only require a simple action like writing an existing dataframe to csv. Rather than calling a function to build the target, you can actually write the code to build the target into the target definition. It can feel a bit clunky for file targets, since you need to return the filepath, but it can be handy for small steps that only need a couple lines of code, e.g., filtering a dataframe object target:
```r
# Save the processed data
  tar_target(
    model_summary_results_csv,
    {
      out_file = "2_process/out/model_summary_results.csv"
      readr::write_csv(eval_data, 
                file = out_file)
      return(out_file)
    }, 
    format = "file"
  ),
```
If it's just a single line/pipe of code, you don't need those curly braces `{ }`:
```r
  tar_target(
    raw_data_filtered,
    readr::read_csv(model_rmses_csv, col_types = cols()) %>%
      filter(exper_n == 1)
  ),
```

Reply:
Oh, cool! Thanks for letting me know about that.